packages/azure/data_stream/waf_logs/elasticsearch/ingest_pipeline/default.yml

```
@@ -0,0 +1,290 @@
+---
+description: Pipeline for processing Azure firewall logs.
+processors:
+  - set:
+      field: ecs.version
+      value: '8.3.0'
+  - set:
+      field: event.kind
+      value: event
+  - set:
+      field: event.category
+      value: [network]
+  - set:
+      field: event.type
+      value: [connection]
+  - rename:
+      field: message
+      target_field: event.original
+      ignore_missing: true
+  - json:
+      field: event.original
+      target_field: json
+      ignore_failure: true
+  - date:
+      field: json.time
+      target_field: '@timestamp'
+      ignore_failure: true
+      formats:
+        - ISO8601
+      if: ctx.json?.time != null
+  - date:
+      field: json.timestamp
+      target_field: '@timestamp'
+      ignore_failure: true
+      formats:
+        - ISO8601
+      if: ctx.json?.time != null
+  - rename:
+      field: json.resourceId
+      target_field: azure.resource_id
+      ignore_missing: true
+  - rename:
+      field: json.properties.clientIP
+      target_field: source.address
+      ignore_missing: true
+  - convert:
+      field: source.address
+      target_field: source.ip
+      type: ip
+      ignore_missing: true
+  - convert:
+      field: json.properties.clientPort
+      target_field: source.port
+      type: long
+      ignore_missing: true
+  - rename:
+      field: json.properties.httpMethod
+      target_field: http.request.method
+      ignore_missing: true
+  - rename:
+      field: json.properties.requestUri
+      target_field: url.path
+      ignore_missing: true
+  - set:
+      field: url.path
+      copy_from: url.original
+      ignore_empty_value: true
+  - rename:
+      field: json.properties.requestQuery
+      target_field: url.query
+      ignore_missing: true
+  - user_agent:
+      field: json.properties.userAgent
+      ignore_missing: true
+      if: ctx.json?.properties?.userAgent != "-"
+  - rename:
+      field: json.properties.httpStatus
+      target_field: http.response.status_code
+      ignore_missing: true
+  - dissect:
+      field: json.properties.httpVersion
+      pattern: "%{}/%{http.version}"
+      ignore_failure: true
+      ignore_missing: true
+  - set:
+      field: network.protocol
+      value: http
+  - rename:
+      field: json.properties.receivedBytes
+      target_field: source.bytes
+      ignore_missing: true
+  - rename:
+      field: json.properties.sentBytes
+      target_field: destination.bytes
+      ignore_missing: true
+  - script:
+      lang: painless
+      source: "ctx.network.bytes = ctx.source.bytes + ctx.destination.bytes"
+      if: ctx.source?.bytes != null && ctx.destination?.bytes != null && ctx.network?.bytes == null
+      ignore_failure: true
+  - script:
+      lang: painless
+      if: ctx.event?.duration != null
+      params:
+        MS_TO_NS: 1000000
+      source: >-
+        ctx.event.duration *= params.S_TO_NS;
```

```suggestion
      params:
        MS_TO_NS: 1000000
      source: >-
        ctx.event.duration *= params.MS_TO_NS;
```
What is the performance impact of using a parameter look-up over a constant with a comment?

---

I don't know, i've always seen a param used.

---

Fair enough.